generator.js

```diff
@@ -61,6 +62,12 @@ module.exports = (api, options) => {
   })
 
   api.onCreateComplete(() => {
+    // early return if cordova binary is not found
```

Comment:
I didn't check, but is onCreateComplete not after postProcessFiles

Reply:
https://github.com/vuejs/vue-cli/blob/9db3c23944b0b326dc62edc81f274510f7e6fa4f/packages/%40vue/cli/lib/GeneratorAPI.js#L208-L224

Reply:
> Also, I didn't try this, but maybe we can already solve it at package manager level...

We could, not sure about peer-dependencies but as a local dependency. In fact there are a few examples of @vuejs plugins that do this, think about `jest`. IMO not depending on global packages is a good thing.

That said, this is not trivial, but I'm willing to give it a shot. Maybe this could be put on hold until I've a better understanding how this could be done.

Reply:
I looked into this, peer dependencies is only showing a warning, but still installing the package, so not optimal.

Looking your code, I prefer, if you do:

```javascript
const checkPrerequisites = () => {
  return hasbin.sync('cordova')
}

module.exports = (api, options) => {
  if (checkPrerequisites()) {
    // current code
  } else {
    api.exitLog('Cordova needs ...')
  }
```

in generator.js

This avoids, that any callbacks are registered or files processed etc...

Should we also check in prompts.js ? Because, if we only modify generator.js, the questions are still asked...

Reply:
@m0dch3n thanks for looking into this. I did put this into hold until I had time to check how to integrate with cordova. 

But truth is we still have the option to improve the PR, merge if all is fine, and then re-evaluate if we finally integrate with cordova.